app/models/theme_field.rb

```diff
@@ -339,7 +339,7 @@ def filename_from_opts(opts)
                          canonical: -> (h) { "locales/#{h[:name]}.yml" }),
     ThemeFileMatcher.new(regex: /(?!)/, # Never match uploads by filename, they must be named in about.json
                          names: nil, types: :theme_upload_var, targets: :common,
-                         canonical: -> (h) { "assets/#{h[:filename]}" }),
+                         canonical: -> (h) { "assets/#{h[:name].gsub(/[^0-9A-Za-z.\-]/, '_')}#{File.extname(h[:filename])}" }),
```

Comment:
I don't see a test here for the filename substitution.

Reply:
I originally added that substitution more as a precaution, because the name of a theme field is already validated [here](https://github.com/discourse/discourse/blob/cfddfa6de2bba1f631cde9054a90bf99910ae2a7/app/models/theme_field.rb#L65).